tests/backend-storage/src/test/ClientStorage.test.ts

```diff
@@ -164,6 +165,40 @@ describe(`${ClientStorage.name}: ${clientStorage.constructor.name}`, () => {
 
         await assertLocalFile(response, contentBuffer);
       });
+
+      it(`should cancel file download from URL`, async () => {
+        const contentBuffer = Buffer.from("test-download-from-url-to-path");
+        const testDownloadPath: string =
+          await testLocalFileManager.getDownloadsDir();
+        const testDirectory: TestRemoteDirectory =
+          await testDirectoryManager.createNew();
+        const uploadedFile: ObjectReference = await testDirectory.uploadFile(
+          { objectName: "file-to-download-from-url.txt" },
+          contentBuffer,
+          undefined
+        );
+
+        const downloadUrl = await serverStorage.getDownloadUrl(uploadedFile);
+        const abortController = new AbortController();
+
+        const downloadPromise = clientStorage.download({
+          url: downloadUrl,
+          transferType: "local",
+          localPath: `${testDownloadPath}/download-url.txt`,
+          abortSignal: abortController.signal,
+        });
+
+        abortController.abort();
+
+        let wasAborted = false;
+        try {
+          await downloadPromise;
+        } catch (error: unknown) {
```

Comment:
More of a question than a comment - is the error thrown different based on the underlying sdk used or does it follow some standards? Could we assert some specific properties here to indicate that the error was thrown due to download being cancelled and not for some other reason?

Reply:
[WHATWG standard](https://webidl.spec.whatwg.org/#aborterror) suggest using `"AbortError"` as `name` value in `Error` object. Although Axios doesn't seem to be following it:

Azure error:
```ts
name: 'AbortError'
message: 'The operation was aborted.'
```

S3 error:
```ts
name: 'AbortError'
message: 'Request aborted'
```

Axios error:
```ts
code: 'ERR_CANCELED'
message: 'canceled'
name: 'CanceledError'
```

This is a problem since I imagine we want object-storage API to return identical errors regardless of underlying SDK.

I see 2 options:

1. Just catch Axios error and rethrow it with property `name` value matching others (`'AbortError'`).
2. Have separate `AbortError` class, catch all thrown abort errors and map then to `AbortError` (this is probably an overkill).

Reply:
I see you have implemented option 1 and I agree that that's the way to go. Looks good.